### ISSUE TITLE
Race between reaper and Go process management

### ISSUE THREAD

**Issue by leoluk** [monogon-dev/monogon#15]
(originally reported by @lorenz in T968)

Hit during tests (<<0.1% probability):

> supervisor                       E0412 12:24:11.249739 supervisor_processor.go:198] Runnable root.enrolment died: returned error when NODE_STATE_NEW: could not make and mount data partition: formatting encrypted block device: wait: no child processes

Likely caused by a race condition between pid1 reaping and Go's command.Run().

**Comment by leoluk** [comment]
Comment by @q3k:

High serverity (thus we need to fix this in MVP) but low probability.

Possible solutions:

a) 'ugly': wait N seconds before reaping (let's not do this)
b) 'nolibc': implement a separate small init before we run node code, in C using the 'nolibc.h' stub-library fron Linux
c) 'rust': implement a separate small init before we run node code, in no_std rust.

We should probably start off with b), even though c) is more attractive, as Rust integration is not very fun with rules_rust as it is (little to no Platform support, but they're on it).

The reason a separate init would help, is that it would only have to reap orphaned child processes, while our Go binary continues to be responsible for its own children started with os/exec.

**Comment by q3k** [comment]
Just hit this again: https://jenkins.monogon.dev/job/gerrit-presubmit-monogon/job/05%252F205%252F2/1/console

```
root.enrolment                   I0707 13:27:09.919585 manager_bootstrap.go:41] Bootstrapping: mounting new storage...
[...]
supervisor                       E0707 13:27:09.943666 supervisor_processor.go:225] Runnable root.enrolment died: returned error when NODE_STATE_NEW: could not make and mount data partition: formatting encrypted block device: wait: no child processes
```

**Comment by q3k** [comment]
And again: https://jenkins.monogon.dev/job/gerrit-presubmit-monogon/job/14%252F314%252F9/1/console

**Comment by q3k** [comment]
Resolved in eac8f73.
